### PR TITLE
fix: Allow KID index '0'

### PR DESCRIPTION
**Pull request by ajupazhamayil**
Fixes https://github.com/firebase/php-jwt/issues/504

### CODE REVIEW

File: src/JWT.php [hunk 1]
```diff
@@ -439,7 +439,7 @@ private static function getKey(
             return $keyOrKeyArray;
         }
 
-        if (empty($kid)) {
+        if (!isset($kid)) {
```
Comment (bshaffer) [inline src/JWT.php:442]: One thing that may be a problem here is that `isset` will return true on an empty string, which would _not_ be a valid $kid.

Comment (ajupazhamayil) [reply src/JWT.php:442]: @bshaffer How about we change it to:
`if (empty($kid) and $kid != '0')`

I don't see any other corner case than this. What do you think?

Comment (bshaffer) [reply src/JWT.php:442]: This seems good to me! `isset` will also return true for `$kid = 0` and `$kid = false`, which I am not sure are valid KIDs

Comment (bshaffer) [reply src/JWT.php:442]: Use `!==` though!

```php
if (empty($kid) and $kid !== '0') {
```

Comment (ajupazhamayil) [reply src/JWT.php]: Yeah right. Done. Thank you @bshaffer for the suggestion. Please review the change.

**Review by bshaffer — APPROVED** [review]

**Pull request merged**